### DISCUSSION TITLE
How to use the TextInput / XKeyPad for long text?

### DISCUSSION THREAD

**Discussion by u1550l** [ImpulseAdventure/GUIslice#514] [Q&A]
Hello,
In GUIslice builder I placed a "TextInput" element on a page and gave a 40 character long string as initial value.
That works perfect.
When I klick the element on the screen, the XKeyPad opens and shows only 18 char and limits input to 18 char.
How can I change this to 40?

Thanks in advance.
Richard

**Comment by Pconti31** [comment]
@u1550l You can try edit of XKeyPad.h 
```
#define XKEYPAD_BUF_MAX  41  // Maximum buffer length for input string
```
It won't change the size of the field you see on screen but should allow scrolling of the input field to 40 characters.

Paul--

**Comment by u1550l** [comment]
Paul,
thanks for the quick answer.
Your suggested patch does part of the work.
I can then edit more :)
But there is no horizontal scrolling. The string is displayed over the right border of the edit field.
![openingTheKeyPad](https://github.com/ImpulseAdventure/GUIslice/assets/31414083/55a71567-5b04-4355-94ff-e8e9fb62ae0f)

you can see it even better after pressing the ">" Right arrow:
![afterPressingRightArrow](https://github.com/ImpulseAdventure/GUIslice/assets/31414083/75791f78-2244-405b-a6d9-4a6797865afb)

There is some issue with fitting in the window and scrolling.

regards - Richard

**Comment by Pconti31** [comment]
@u1550l Yes, the scrolling doesn't seem to work with the Alpha keyboard.   Only the numeric version is scrolling.  
The numeric version doesn't seem to need XKEYPAD_BUF_MAX  changed. You will need to wait for Calvin to patch the alpha version or dig into the code yourself.  I have no idea why its not scrolling.  The original keyboard I summitted for inclusion into guislice didn't need max setting except to describe the viewing field.  Max limit was simply the size of the field you were reading into.   
Paul--

**Comment by u1550l** [comment]
Paul,
thanks for your help.
I will have a look into the code...
-- Richard --

**Comment by u1550l** [comment]
Paul,
I think, I made a mistake: While trying myself to solve my question, I had changed XKEYPAD_DISP_MAX in XKeyPad_Alpha-setup.h from 18 to 40 and forgotten to change it back. This explains the long text field.
The TextInput now works perfectly. 
Thanks again for your help.
-- RIchard --
